app/src/main/java/com/android/gatherly/model/todo/TodosRepositoryFirestore.kt

```diff
@@ -183,8 +186,35 @@ class ToDosRepositoryFirestore(private val db: FirebaseFirestore) : ToDosReposit
     val ownerId = doc.getString("ownerId") ?: return null
     val statusStr = doc.getString("status") ?: ToDoStatus.ONGOING.name
     val status = ToDoStatus.valueOf(statusStr)
+    val priorityStr = doc.getString("priority") ?: return null
+    val priority = ToDoPriority.valueOf(priorityStr)
+    val categoryMap = doc.get("tag") as? Map<*, *>
```

Comment:
```suggestion
    val categoryMap = doc["tag"] as? Map<*, *>
```

I believe that this should help you fix the sonar warning